app/services/time_discrepancy_loader.rb

```diff
@@ -2,24 +2,58 @@
 
 class TimeDiscrepancyLoader
   DECIMALS_TO_ROUND_TO = 2
+  BASE_SELECT = 'timer_sessions.*'
+  SELECT_SEPARATOR = ','
+  VIRTUAL_ATTRIBUTE = 'discrepancy_detected'
 
-  # base_query ==> Expected to be TimerSession.where(user_id: user.id)
-  def initialize(base_query)
-    @base_query = base_query
+  def initialize(timer_sessions)
+    @timer_sessions = timer_sessions
+    @timer_session_arel = TimerSession.arel_table
+    @time_entry_arel = TimeEntry.arel_table
   end
 
-  def where_time_not_adding_up
-    build_query
+  def load_with_discrepancy_state
```

Comment:
has to done this way due to a bug in rails https://github.com/rails/rails/issues/15185